src/main/java/com/google/sps/servlets/CreateGameServlet.java

```diff
@@ -40,38 +40,117 @@
 */
 @WebServlet("/create-game-data")
 public class CreateGameServlet extends HttpServlet {
+    MockDatastoreManager datastoreManager = new MockDatastoreManager();
     Gson gson = new Gson();
+    String gameID;
+    String userID;
+    String gameName;
+    String gameDescription;
+    ArrayList<String> stageKeys;
+    ArrayList<Coordinates> stageSpawnLocations;
+    ArrayList<String> stageStarterHints;
+    ArrayList<ArrayList<Coordinates>> hintLocations;
+    ArrayList<ArrayList<String>> hintTexts;
 
+    /**
+    * Creates a game with the given fetch parameters.
+    */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        if(!isInputValid(request)) {
+        parseInput(request);
+        if(!isInputValid()) {
             System.out.println("Input json is not valid");
             System.exit(1);
         }
-        String gameID = IDGenerator.gen();
+
+        gameID = IDGenerator.gen();
+
+        Game.Builder gameBuilder = new Game.Builder(gameID, gameName);
+        gameBuilder.setGameCreator(userID);
+        gameBuilder.setGameDescription(gameDescription);
+        ArrayList<String> stages = new ArrayList<>();
+        for(int i = 0; i < stageKeys.size(); i++) {
+            Stage stage = buildStage(i);
+            stages.add(stage.getStageID());
+            datastoreManager.storeStage(stage);
+        }
+        gameBuilder.setStages(stages);
+        gameBuilder.setNumTimesPlayed(0);
+        gameBuilder.setNumTimesFinished(0);
+        gameBuilder.setNumStarVotes(0);
+        gameBuilder.setTotalStars(0);
+        gameBuilder.setNumDifficultyVotes(0);
+        gameBuilder.setTotalDifficulty(0);
+        Game game = gameBuilder.build();
+        datastoreManager.storeGame(game);
+
         response.getWriter().println(gameID);
     }
 
     /**
-    * Checks that the values/sizes of numStages, numHints, stageSpawnLocations, stageStarterHints,
-    * hintLocations, and hintTexts all are consistent with one another.
+    * Creates a Hint object given the stage index and the hint index.
+    * stageIdx = 0 corresponds to the 1st stage, and hintIdx = 0
+    * corresponds to the 1st hint.
+    * @param stageIdx the stage index.
+    * @param hintIdx the hint index within the stage.
```

Comment:
"Idx" is unclear what it is, try to just put index to make it more clear that it is the index.

Reply:
Changed, thanks!